confmap/provider/s3provider/provider.go

```diff
@@ -44,6 +44,27 @@ type provider struct {
 //
 // Examples:
 // `s3://DOC-EXAMPLE-BUCKET.s3.us-west-2.amazonaws.com/photos/puppy.jpg` - (unix, windows)
+func NewFactory() confmap.ProviderFactory {
```

Comment:
The comment above this method needs to be updated to say `NewFactory` instead of `New`, and the return type should be updated as well.

Reply:
Doh! Updated, thanks for catching that.